### PR TITLE
Tabbed panels, form generation mechanics and support for simple fields

### LINKED ISSUE #35: Add a Request form
In addition to the text area (the current implementation) we need a form to edit the aggregation inputs to avoid having to implement a code editor with suggestions.

This issue focuses on simpler inputs, wheret the aggregation argument is a number, bool, string, or date.

For arguments being objects or arrays, if the implementation is less than trivial we can open a new issue.

### PR DESCRIPTION
**Pull request by saabi**
Closes #35

### CODE REVIEW

File: src/routes/builder/index.svelte [hunk 1]
```diff
@@ -132,25 +132,40 @@
...
+	import TabContainer from 'app/components/elementary/TabContainer.svelte';
+	import Tab from 'app/components/elementary/Tab.svelte';
+	import Select from 'app/components/elementary/Select.svelte';
+	import SelectMenu from 'app/components/elementary/SelectMenu.svelte';
 	import PanelMenu from 'app/components/elementary/PanelMenu.svelte';
 	import MenuItem from 'app/components/elementary/MenuItem.svelte';
 	import IconDelete from 'app/components/icons/IconDelete.svelte';
 
+	let datasetTypings;
 	const AXIS_NAMES = ['primary', 'secondary', 'tertiary', 'quaternary', 'quinary', 'senary', 'septenary', 'octonary', 'nonary', 'denary'];
```
Comment (mindrones) [inline src/routes/builder/index.svelte]: As a note for later as this is not in scope, I'd generate this at run-time like [`agg-1`, `agg-2`, ...], no need to pre-compute 10 keys.

File: src/routes/builder/index.svelte [hunk 2]
```diff
@@ -181,24 +196,77 @@
...
+			size: resultSize
 		}
 	}
+	function clearParameters () {
+		axisParams[selectedAxis].input = {};
+	}
 
 	const isMissing = (key, value) => obj => Boolean(obj) && !obj[key].has(value);
 
+	function computeRequestBody (config) {
```
Comment (mindrones) [inline src/routes/builder/index.svelte:241]: just noting that we agreed to refactor this function in another PR

File: src/node_modules/app/components/elementary/TabContainer.svelte [hunk 3]
```diff
@@ -0,0 +1,55 @@
+<script context="module">
+	export const tabsKey = {};
+</script>
+
+<script>
+	// eslint-disable-next-line node/no-unpublished-import
```
Comment (mindrones) [inline src/node_modules/app/components/elementary/TabContainer.svelte:6]: I'd disable at module level

File: src/node_modules/app/components/elementary/ElasticSearchField.svelte [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+		}
+		return newDataType;
+	}
+
+	$: placeholder = computePlaceHolder(dataType);
+	$: {
+		fieldValue = formatIncomingValue(value);
+		currentValue = fieldValue;
+	}
+	$: type = computeInputType(dataType);
```
Comment (mindrones) [inline src/node_modules/app/components/elementary/ElasticSearchField.svelte]: not sure we need to invoke a function here, I'd just use

`$: placeholder = inputTypes[esType];`

File: src/node_modules/app/components/elementary/ElasticSearchField.svelte [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+	>
+{:else if dataType === 'boolean'}
+<div>
+	<input id={`${id}-unset`} type='radio' bind:group={fieldValue} value={'null'}>
+	<label for={`${id}-unset`} class='boolean-state'>Unset</label>
+	<input id={`${id}-true`} type='radio' bind:group={fieldValue} value={true}>
+	<label for={`${id}-true`} class='boolean-state'>True</label>
+	<input id={`${id}-false`} type='radio' bind:group={fieldValue} value={false}>
+	<label for={`${id}-false`} class='boolean-state'>False</label>
+</div>
```
Comment (mindrones) [inline src/node_modules/app/components/elementary/ElasticSearchField.svelte]: Possibly I'd make this into a component

**Review by mindrones — APPROVED** [review]
I left some minor notes below and as discussed we'll need to refactor some of this, but so far great job!

Good to merge!

File: src/routes/builder/index.svelte (hunk 1 above)
Comment (saabi) [reply src/routes/builder/index.svelte]: Actually, this should forcibly change when we make aggregations independent.

**Pull request merged**